src/analysis/processing/qgsalgorithmvoronoipolygons.cpp

```diff
@@ -204,11 +204,38 @@ QString QgsVoronoiPolygonsAlgorithm::voronoiWithoutAttributes( const QVariantMap
   if ( !sink )
     throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
 
+  QgsGeometry allPoints;
```

Comment:
It'd be more efficient to make this a `std::unique_ptr< QgsMultiPoint > ` instead, and call addGeometry on the multipoint directly instead of calling QgsGeometry::addPart (which has a LOT of extra stuff going on!).

Reply:
Should be fixed, thanks.